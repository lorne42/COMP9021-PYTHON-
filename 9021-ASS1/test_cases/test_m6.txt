

One day, you stumble upon a hidden village known only to the inhabitants as The Land of Truth and Deception. 
In this village, there are two types of people: Knights, who always tell the truth, and Knaves, who always lie. 
You encounter a group of villagers and decide to determine who is a Knight and who is a Knave.
0011001
Sir Alpha proclaims: "All of us are Knaves."   
"Exactly one of us is a Knight," retorts Sir Beta.
Sir Gamma whispers: "At least one of Sir Alpha and I is a Knave."                  
"At most one of Sir Beta and I is a Knight," claims Sir Delta.
Sir Epsilon says: "Exactly one of Sir Gamma, Sir Delta, and I is a Knight!"
"Sir Alpha or Sir Beta is a Knight," exclaims Sir Zeta.             
"Sir Gamma and Sir Delta are Knights," asserts Sir Eta.                   
What's your opinion?



