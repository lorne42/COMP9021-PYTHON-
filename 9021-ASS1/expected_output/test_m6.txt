The Sirs are: Alpha Beta Delta Epsilon Eta Gamma Zeta
There is a unique solution:
Sir Alpha is a Knave.
Sir Beta is a Knave.
Sir Delta is a Knight.
Sir Epsilon is a Knave.
Sir Eta is a Knight.
Sir Gamma is a Knight.
Sir Zeta is a Knave.